modules/dcache/src/main/java/org/dcache/services/ssh2/Ssh2Admin.java

```diff
@@ -318,6 +327,66 @@ public boolean authenticate(String userName, PublicKey key,
         }
     }
 
+    private class RemoteHostValidator {
+
+        private boolean isValidHost(String line, ServerSession session) {
```

Comment:
I have updated the method to make a decision based on a combination of all entries, as you suggested below. Testing with multiple combinations on my home network I was able to confirm that all the cases you mentioned are now handled correctly. I will write some unit tests and open a new PR for them